packages/sdk/src/sdk/services/AudiusWalletClient/createHedgehogWalletClient.ts

```diff
@@ -0,0 +1,91 @@
+import type { Hedgehog } from '@audius/hedgehog'
+import {
+  createClient,
+  custom,
+  CustomTransport,
+  type Hex,
+  type SignMessageParameters,
+  type SignTypedDataParameters,
+  type TypedData
+} from 'viem'
+import { getAddresses, signMessage, signTypedData } from 'viem/actions'
+
+import {
+  getSharedSecret,
+  type GetSharedSecretParameters
+} from './actions/getSharedSecret'
+import { sign, type SignParameters } from './actions/sign'
+import { localTransport } from './localTransport'
+import { privateKeyToAudiusAccount } from './privateKeyToAudiusAccount'
+import type { AudiusAccount, AudiusWalletClient } from './types'
+
+/**
+ * Creates a Viem client that uses a local Hedgehog instance to do all the wallet methods.
+ */
+export function createHedgehogWalletClient(
+  hedgehog: Hedgehog
+): AudiusWalletClient {
+  // Gets the local privateKeyAccount for the Hedgehog wallet.
+  // Unfortunately, Hedgehog loads the wallet asynchronously from storage,
+  // so we can't have the account already initialized on start.
+  let account_: AudiusAccount | undefined
+  const getAccount = async () => {
+    await hedgehog.waitUntilReady()
+    const wallet = hedgehog.getWallet()
+    if (!wallet) {
+      throw new Error('Hedgehog wallet not found. Is the user logged in?')
+    }
+
+    if (!account_ || account_.address !== wallet.getAddressString()) {
+      account_ = privateKeyToAudiusAccount(wallet.getPrivateKeyString() as Hex)
+    }
+    return account_
+  }
+
+  // Init the client as though we have an account
+  const client = createClient<CustomTransport, undefined, AudiusAccount>({
+    name: 'hedgehog',
+    type: 'audius',
+    transport: custom(localTransport())
+  })
+
+  // Set the account lazily, without making this method async
+  getAccount()
+    .then((account) => {
+      client.account = account
+    })
+    .catch(() => {
+      // Do nothing - to be expected if the user is not logged in yet.
```

Comment:
Just wondering how well this plays with the ephemeral wallet pattern we use for anonymous users. I think hedgehog starts un-initialized for a bit and then we call the method to generate a new wallet from the dummy entropy.

Reply:
Me too! I haven't played around much yet with the non-logged in => logged in scenario. Curious to see how this plays with that